src/Compile-SourceScript/Public/Compile-SourceScript.ps1

```diff
@@ -55,8 +59,14 @@ function Compile-SourceScript {
             if (!$MOD_NAME) {
                 throw "File is not a '.sp' or '.sma' source file."
             }
-            if (!($sourceFile.DirectoryName | Split-Path)) {
-                throw "The directory 'addons/$MOD_NAME/' cannot exist relative to the specified plugin source file '$($sourceFile.FullName)'."
+            if ($ScriptingDirectory) {
+                if (!(Test-Path $ScriptingDirectory -PathType Container)) {
+                    throw "Specified scripting directory $ScriptingDirectory must be an existing directory."
+                }
+            }else {
+                if (!($sourceFile.DirectoryName | Split-Path)) {
```

Comment:
This check is mandatory and so should be performed in any case, not just if the `scripting` directory is specified by the user. Either duplicated prior to processing global variables or performed on the variable `$SCRIPTING_DIR` after it has been assigned.

Reply:
Fixed

Reply:
This isn't yet resolved correctly. The check must be performed on the scripting directory whether the scripting directory is specified or not via `-ScriptingDirectory`. This ensures the scripting directory 

Actually this check was added to ensure that the user is developing within the mod's default `scripting` directory. It ensures the specified plugin source file is exists within a directory, just never within the `/` directory, since the relative path of the directory `plugins` must be valid for installation of the plugin after successful compilation(s).